File: 백준/Gold/1711. 직각삼각형/직각삼각형.java
import java.awt.*;
import java.io.BufferedReader;
import java.io.FileInputStream;
import java.io.InputStreamReader;
import java.util.*;
import java.util.List;


public class Main {

    public static void main(String[] args) throws Exception {
        //System.setIn(new FileInputStream("./src/input.txt"));
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int size = Integer.parseInt(br.readLine());
        long pxList[] = new long[size];
        long pyList[] = new long[size];
        for (int i = 0; i < size; i++) {
            StringTokenizer stk = new StringTokenizer(br.readLine());
            long x = Long.parseLong(stk.nextToken());
            long y = Long.parseLong(stk.nextToken());
            pxList[i] = x;
            pyList[i] = y;
        }

        int ans = 0;
        for (int a = 0; a < size - 2; a++) {
            for (int b = a + 1; b < size - 1; b++) {
                for (int c = b + 1; c < size; c++) {
                    long ax = pxList[a];
                    long ay = pyList[a];
                    long bx = pxList[b];
                    long by = pyList[b];
                    long cx = pxList[c];
                    long cy = pyList[c];

                    long aTob = (ax-bx)*(ax-bx) + (ay-by)*(ay-by);
                    long aToc = (ax-cx)*(ax-cx) + (ay-cy)*(ay-cy);
                    long bToc = (bx-cx)*(bx-cx) + (by-cy)*(by-cy);
                    if(aTob + aToc == bToc || aToc + bToc == aTob || aTob + bToc == aToc){
                        ans ++;
                    }



                }
            }
        }
        System.out.println(ans);

    }

}

File: 백준/Gold/1711. 직각삼각형/직각삼각형.py
import sys
size = int(sys.stdin.readline())
px_list = []
py_list = []
for _ in range(size):
    x, y = map(int, sys.stdin.readline().split())
    px_list.append(x)
    py_list.append(y)
ans = 0
for a in range(size - 2):
    for b in range(a + 1, size - 1):
        for c in range(b + 1, size):
            ax, ay = px_list[a], py_list[a]
            bx, by = px_list[b], py_list[b]
            cx, cy = px_list[c], py_list[c]
            a_to_b = (ax - bx) ** 2 + (ay - by) ** 2
            a_to_c = (ax - cx) ** 2 + (ay - cy) ** 2
            b_to_c = (bx - cx) ** 2 + (by - cy) ** 2
            if a_to_b + a_to_c == b_to_c or a_to_c + b_to_c == a_to_b or a_to_b + b_to_c == a_to_c:
                ans += 1
print(ans)